.github/workflows/pr_build.yaml

```diff
@@ -76,6 +76,25 @@ jobs:
       - name: Run unit tests
         run: ./.github/workflows/scripts/run_unit_tests.sh
 
+  unit-test-race-detector:
+    name: Run unit tests with race detector (linux)
```

Comment:
I've cycled on the name here a few times. I'm not happy with the current. I would love suggestions/guidance.